Description: Cherry-picked patch from upstream to close RestoreForState querier.
 See https://github.com/prometheus/prometheus/pull/4922
--- a/rules/manager.go
+++ b/rules/manager.go
@@ -599,6 +599,7 @@
 
 		alertRule.SetRestored(true)
 	}
+	defer q.Close()
 
 }
 
